bitbots_time_constraint/src/bitbots_time_constraint/check_constraints.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python3
+# -*- coding:utf-8 -*-
+
+import rospy
+import rostopic
+import rospkg
+import yaml
+import os
+
+class AutoSubscriber:
+    """
+    A class which automatically subscribes to a topic as soon as it becomes available and saves the time when messages come in.
+    """
+
+    def __init__(self, topic):
+        """
+        :param topic: Topic to subscribe to
+        :type topic: str
+        """
+        self.topic = topic
+        self.latest_timestamp = None
+
+        self.__subscriber = None        # type: rospy.Subscriber
+        self.__subscribe()
+
+    def __subscribe(self, backoff=1.0):
+        """
+        Try to subscribe to the set topic
+        :param backoff: How long to wait until another try
+        """
+        data_class, _, _ = rostopic.get_topic_class(self.topic)
+        if data_class is not None:
+            # topic is known
+            self.__subscriber = rospy.Subscriber(self.topic, data_class, self.__message_callback, queue_size=1, tcp_nodelay=True)
+            rospy.loginfo('Subscribed to topic {}'.format(self.topic))
+
+        else:
+            # topic is not yet known
+            rospy.loginfo('Topic {} is not yet known. Retrying in {} seconds'.format(self.topic, int(backoff)))
+            rospy.Timer(
+                rospy.Duration(int(backoff)),
+                lambda event: self.__subscribe(backoff * 1.2),
```

Comment:
why ```* 1.2```?

Reply:
Its recursively called and with each call, the retry time increases with this magic value.